tools/lib/isGzippedFile.js

```diff
@@ -1,15 +1,24 @@
 'use strict';
 var fsExtra = require('fs-extra');
+var Promise = require('bluebird');
 var isGzipped = require('./isGzipped');
 
 module.exports = isGzippedFile;
 
+var readStreamOptions = {
+    start : 0,
+    end : 2
+};
+
 /**
  * @private
  */
 function isGzippedFile(file) {
-    return fsExtra.readFile(file)
-        .then(function (buffer) {
-            return isGzipped(buffer);
+    return new Promise(function (resolve, reject) {
+        var readStream = fsExtra.createReadStream(file, readStreamOptions);
+        readStream.on('error', reject);
+        readStream.on('data', function(chunk) {
```

Comment:
This is just checking the first chunk right? Should this be `readStream.once('data' //...`? Also, I'm still not 100% with streams, but should this get manually closed after the first chunk is read?